File: projects/KD1OUR/automata/automata-java/src/statemachine/java/model/TerminalSymbol.java
package statemachine.java.model;

/**
 * A terminális szimbúlumok fajtája, ahogy a feladatleírásban szerepelnek.
 */
public enum TerminalSymbol {
    DIGIT, POINT, SIGN, ZERO, E;

    /**
     * Egy karakterhez hozzárendeli a terminális szimbólum típusát.
     * @param character
     *      bemeneti karakter
     * @return
     *      terminális szimbólum típus, amennyiben egy érvényes terminális szimbólum, egyébként <code>null</code>
     */
    public static TerminalSymbol valueOf(char character){
        if(character == '0'){
            return TerminalSymbol.ZERO;
        } else if(Character.isDigit(character)){
            return TerminalSymbol.DIGIT;
        } else if (character == '.') {
            return TerminalSymbol.POINT;
        } else if (character == '+' || character == '-') {
            return TerminalSymbol.SIGN;
        } else if (character == 'e' || character == 'E') {
            return TerminalSymbol.E;
        }

        return null;
    }
}

File: projects/KD1OUR/automata/automata-scala/src/statemachine/scala/model/TerminalSymbol.scala
package statemachine.scala.model

/**
 * A terminális szimbúlumok fajtája, ahogy a feladatleírásban szerepelnek.
 */
object TerminalSymbol extends Enumeration {

  type TerminalSymbol = Value

  val DIGIT, POINT, SIGN, ZERO, E = Value

  /**
   * Egy karakterhez hozzárendeli a terminális szimbólum típusát.
   * @param char
     * bemeneti karakter
   * @return
     * terminális szimbólum típus, amennyiben egy érvényes terminális szimbólum, egyébként <code>null</code>
   */
  def getTerminalSymbol(char : Char) : Option[TerminalSymbol] = {
    if(char == '0') Some(ZERO)
    else if (char.isDigit) Some(DIGIT)
    else if (char == '.') Some(POINT)
    else if (char == '+'  || char == '-') Some(SIGN)
    else if (char == 'e' || char == 'E') Some(E)
    else None
  }

}
